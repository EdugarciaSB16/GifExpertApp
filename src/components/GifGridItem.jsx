import React from 'react';


const GifGridItem = ({ title, url }) => {
    return (
        <div className='card animate__animated animate__fadeInDown'>
            <div>
                <img src={url} alt={title} />
            </div>
            <div>
                <span>{title}</span>
            </div>
        </div>
    );
};

export default GifGridItem;