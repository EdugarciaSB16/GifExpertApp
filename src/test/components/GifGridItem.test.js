import React from 'react'
import GifGridItem from '../../components/GifGridItem'
import { shallow } from 'enzyme'

describe('Pruebas al componente GifGridItem', () => {
    
    const title = 'titulo'
    const url = 'https://ejemplo.jpg'
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>)

    test('Debe de renderizar el componente', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe de tener un parrafo con title', () => {

        const parrafo = wrapper.find('span')
        expect( parrafo.text().trim() ).toBe( title )

    })  

    test('Debe de tener una imagen con su alt', () => {

        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
    })

    test('Debe de tener las clases animate', () => {

        const div = wrapper.find('.card')
        const className = div.prop('className')
        expect(className.includes('animate__fadeInDo')).toBe(true)
    })
})