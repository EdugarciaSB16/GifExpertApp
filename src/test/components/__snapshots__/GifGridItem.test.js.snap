// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGridItem Debe de renderizar el componente 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <div>
    <img
      alt="titulo"
      src="https://ejemplo.jpg"
    />
  </div>
  <div>
    <span>
      titulo
    </span>
  </div>
</div>
`;
